Include/object.h

```diff
@@ -461,25 +461,30 @@ static inline void _Py_INCREF(PyObject *op)
 
 #define Py_INCREF(op) _Py_INCREF(_PyObject_CAST(op))
 
+#ifdef Py_REF_DEBUG
 static inline void _Py_DECREF(const char *filename, int lineno,
                               PyObject *op)
 {
-    (void)filename; /* may be unused, shut up -Wunused-parameter */
-    (void)lineno; /* may be unused, shut up -Wunused-parameter */
-    _Py_DEC_REFTOTAL;
-    if (--op->ob_refcnt != 0) {
-#ifdef Py_REF_DEBUG
-        if (op->ob_refcnt < 0) {
-            _Py_NegativeRefcount(filename, lineno, op);
-        }
-#endif
+    _Py_RefTotal--;
```

Comment:
So this _Py_DEC_REFTOTAL would be converted in your plan?

Reply:
_Py_DEC_REFTOTAL = _Py_RefTotal-- when Py_REF_DEBUG is defined. I prefer to put directly the real code, it may help debugging. It avoids the indirection of the preprocessor.